### PR TITLE
247 Improve Server Errors

### LINKED ISSUE #247: Improve Server Errors
Improve erros to adhere [Problem Schema](https://zalando.github.io/problem/schema.yaml#/Problem).

- Improve `error.detail` so that they can be used as it is (as a message to user) on the front-end.
- Use appropriate `error.status`.

### PR DESCRIPTION
**Pull request by netme**
Closes #247 

Fixes issue introduced in #217

### CODE REVIEW

File: server/src/test/java/de/zalando/zally/RestApiViolationsTest.java [hunk 1]
```diff
@@ -205,4 +234,18 @@ private String getLocalUrl(final String resourceFilePath, final String contentTy
...
+    private String simulateNotFound() throws Exception {
+        final String remotePath = "/abcde.yaml";
+        final String url = "http://localhost:" + port() + remotePath;
+
+        onRequest()
+                .havingMethodEqualTo("GET")
+                .havingPathEqualTo(remotePath)
+                .respond()
+                .withStatus(404)
+                .withHeader("Content-Type", "text/plain")
```
Comment (fmueller) [inline server/src/test/java/de/zalando/zally/RestApiViolationsTest.java]: Why not problem json content type or application json?

Comment (netme) [inline server/src/test/java/de/zalando/zally/RestApiViolationsTest.java]: Good point, you made me think about adding a problem-json assert to the test which uses this utility call. Will do it soon :-)

Comment (fmueller) [inline server/src/test/java/de/zalando/zally/RestApiViolationsTest.java]: Ok :-)

Comment (netme) [inline server/src/test/java/de/zalando/zally/RestApiViolationsTest.java]: Fixed in e83e6a5

File: server/src/main/java/de/zalando/zally/exception/ExceptionHandling.java [hunk 2]
```diff
@@ -17,4 +20,17 @@
...
     }
+
+    @ExceptionHandler
+    public ResponseEntity<Problem> handleHttpClientErrorException(
+            HttpClientErrorException exception, NativeWebRequest request) {
+        return create(exception.getStatusCode(), exception, request);
+    }
+
+    @ExceptionHandler
+    public ResponseEntity<Problem> handleResourceAccessException(
```
Comment (fmueller) [inline server/src/main/java/de/zalando/zally/exception/ExceptionHandling.java]: By this you are changing error handling for all 404 status codes - also for other resources. I propose to handle the 404 for api-definition-url separately in controller method itself.

Comment (netme) [inline server/src/main/java/de/zalando/zally/exception/ExceptionHandling.java]: OK, got it . Will change it

Comment (netme) [inline server/src/main/java/de/zalando/zally/exception/ExceptionHandling.java]: Done in b49889d

File: server/src/main/java/de/zalando/zally/exception/UnaccessibleResourceUrlException.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package de.zalando.zally.exception;
+
+import org.springframework.http.HttpStatus;
+
+public class UnaccessibleResourceUrlException extends RuntimeException {
+    final private HttpStatus httpStatus;
```
Comment (fmueller) [inline server/src/main/java/de/zalando/zally/exception/UnaccessibleResourceUrlException.java]: Order should be `private final`.

File: server/src/main/java/de/zalando/zally/ApiDefinitionReader.java [hunk 4]
```diff
@@ -12,7 +16,7 @@ public ApiDefinitionReader(JsonNode requestBody) {
         this.requestBody = requestBody;
     }
 
-    public String read() throws MissingApiDefinitionException {
+    public String read() throws RuntimeException {
```
Comment (fmueller) [inline server/src/main/java/de/zalando/zally/ApiDefinitionReader.java]: Don't throw `RuntimeException` if you know only two exceptions are thrown. You can chain different exceptions by comma.

**Comment by netme** [comment]
@fmueller , good catch. Fixed.

**Comment by fmueller** [comment]
👍

**Pull request merged**